processing/src/test/java/org/apache/druid/query/QueryRunnerTestHelper.java

```diff
@@ -467,6 +478,87 @@ public String toString()
     };
   }
 
+  public static <T, QueryType extends Query<T>> List<QueryRunner<T>> makeUnnestQueryRunners(
+      QueryRunnerFactory<T, QueryType> factory,
+      String dimensionToUnnest,
+      String outputColumn,
+      LinkedHashSet<String> allowSet
+  )
+  {
+    final IncrementalIndex rtIndex = TestIndex.getIncrementalTestIndex();
+    final IncrementalIndex noRollupRtIndex = TestIndex.getNoRollupIncrementalTestIndex();
+    final QueryableIndex mMappedTestIndex = TestIndex.getMMappedTestIndex();
+    final QueryableIndex noRollupMMappedTestIndex = TestIndex.getNoRollupMMappedTestIndex();
+    final QueryableIndex mergedRealtimeIndex = TestIndex.mergedRealtimeIndex();
+    final QueryableIndex frontCodedMappedTestIndex = TestIndex.getFrontCodedMMappedTestIndex();
```

Comment:
I'm confused as to why you have to do this?  The Unnest is defined as part of the query, not as part of the query runner, you should be able to just send the query along and it will attach the unnest on the segment via the segment mapper functions, no?